maxwell3d: Implement Conditional Rendering

FernandoS27 opened the pull request:
Conditional Rendering takes care of conditionaly clearing or drawing
depending on a set of queries. This PR implements the query checks to
stablish if things can be rendered or not.

Hexagon12 commented on src/video_core/engines/maxwell_3d.h:
```diff
@@ -89,13 +89,25 @@ class Maxwell3D final {
 
         enum class QuerySelect : u32 {
             Zero = 0,
+            TimeElapsed = 2,
```
Please list these in order. I've also heard that query mode 26 is also known (transform feedback related), add that too.

ReinUsesLisp replied:
I'm fine with just ordering it. We don't exactly know what query 26 is and if we can stub it or not.

ReinUsesLisp commented on src/video_core/engines/maxwell_3d.cpp:
```diff
@@ -342,6 +347,43 @@ void Maxwell3D::ProcessQueryGet() {
+        break;
+    }
+    case Regs::ConditionMode::Equal: {
+        std::array<u32, 6> data;
```
If we know what these values mean, this should be passed as a structure.

FernandoS27 replied:
we kinda don't know them, nouveau uses that but I can have a better look.

FernandoS27 commented:
Ok I've addressed the comments. Changes: comparison now uses a struct instead of an array of data from what I understood they were in nouveau, I ordered the query select types and added query 26 which is related to transform feedback.
